### DISCUSSION TITLE
Confusion about the Code of esm

### DISCUSSION THREAD

**Discussion by Yusuno** [evolutionaryscale/esm#59] [Q&A]
Since attention_mask is already used as the input of the model (StructureTokenEncoder), and the affine_mask is also obtained in the forward process of the model, why do we need to input sequence_id and chain_id to ensure that the model masks irrelevant parts when calculating attention? I am also confused about the format of sequence_id and chain_id. Will each residue have the same id? Will the position that does not need to be paid attention to be 0?
